lib/src/logger.dart

```diff
@@ -174,7 +174,7 @@ class Logger {
     Object? object;
     if (isLoggable(logLevel)) {
       if (message is Function) {
-        message = message();
+        message = (message as Object Function())();
```

Comment:
Previously we would have gracefully handled a `null` return. I'm not sure if anyone is doing that, but I don't see any reason to specifically disallow it.
```suggestion
        message = (message as Object? Function())();
```

Reply:
Ah, good point - `Object Function()` would change the runtime behavior. I'll update this to `Object? Function()`.